frame/staking/src/benchmarking.rs

```diff
@@ -131,6 +132,146 @@ pub fn create_validator_with_nominators<T: Config>(
 	Ok((v_stash, nominators))
 }
 
+struct RebagScenario<T: Config> {
```

Comment:
My fundamental issue with this start the moment I see the word `bag` in `pallet-staking`. Further down I also see notions of `threshold` which is again OT in staking.  

In an ideal world and in my head, staking should only work and talk to `SortedListProvider`. I might be oversimplifying it, but I think the majority of the staking benchmark for something like `bond_extra` should stay the same, and it should only call `T::SortedListProvider::prepare_worse_case(who, amount)`, which means "I need you to make sure a rebag (worse case) happens if I change backing of `who` by `amount`". 

Then, in the receiving side, we make sure that this rebag happens.

Thinking a bit more, it might be easier to do this: 

- Staking will call `T::SortedListProvider::weight_update_worse_case(who)`, in which bags-list will tell staking by how much it should increase `who`'s stake to cause a worse case (aka. rebag).

And a similar analogy should exist in the rest of them as well. 

---

That being said, the benchmarks seem to be correct and solid in terms of delivering worse-case code execution, so I think doing this refactor in a follow-up is also acceptable.